### ISSUE TITLE
Typescript services crash when opening JSX tags

### ISSUE THREAD

**Issue by ChuckJonas** [Microsoft/vscode#49170]
- VSCode Version: 1.22.2
- OS Version: osx 10.12.6

Steps to Reproduce:
I can't figure out the exact steps to reproduce, but it always happens when I'm editing JSX and typically happens more often once the JSX gets bigger and more complex. It most commonly occurs when I open a new tag. 

When there should just be a single `Expected corresponding JSX closing tag` error, instead, hundreds how errors are thrown and everything typescript service related stops working.  No formatting, goto-def, code complete, errors don't go away no matter what I type, etc.  

Simply restarting `ts server` does not fix it.  Have to reload the program, which is annoying because I typically have webpack dev server running in the vscode terminal.

<img width="545" alt="contactroletable_tsx_ _ids-contract-role-mgmt" src="https://user-images.githubusercontent.com/5217568/39612479-2279aad6-4f1d-11e8-9b0b-ada0146d8ab9.png">

Does this issue occur when all extensions are disabled?: Yes/No
Yes (I think... need to double check but it happens very random so it's hard to reproduce)

Output from Typescript Console

```
[Error  - 9:34:06 PM] 'signatureHelp' request failed with error.
Error processing request. Debug Failure. Did not expect JsxExpression to have an Identifier in its trivia
Error: Debug Failure. Did not expect JsxExpression to have an Identifier in its trivia
    at NodeObject.addSyntheticNodes (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:83665:34)
    at NodeObject.createChildren (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:83729:22)
    at NodeObject.getChildren (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:83749:22)
    at getTokenAtPositionWorker (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:66283:43)
    at getTokenAtPosition (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:66274:16)
    at Object.findTokenOnLeftOfPosition (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:66308:31)
    at Object.getSignatureHelpItems (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:75060:36)
    at Object.getSignatureHelpItems (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:84729:37)
    at IOSession.Session.getSignatureHelpItems (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:91641:62)
    at Session.handlers.ts.createMapFromTemplate._a.(anonymous function) (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:90663:61)
    at /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:92067:88
    at IOSession.Session.executeWithRequestId (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:92058:28)
    at IOSession.Session.executeCommand (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:92067:33)
    at IOSession.Session.onMessage (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:92087:35)
    at Interface.<anonymous> (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:93288:27)
    at emitOne (events.js:96:13)
    at Interface.emit (events.js:191:7)
    at Interface._onLine (readline.js:241:10)
    at Interface._normalWrite (readline.js:384:12)
    at Socket.ondata (readline.js:101:10)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:191:7)
    at readableAddChunk (_stream_readable.js:178:18)
    at Socket.Readable.push (_stream_readable.js:136:10)
    at Pipe.onread (net.js:560:20)
```

**Comment by mjbvz** [comment]
Can you please test this in VS Code insiders. We've fixed a number of bugs since 1.22 but I do not know if this one specifically has been fixed or not

Also please let me know if you can determine steps to reliably reproduce the problem

**Comment by mjbvz** [comment]
No, I do not think the standard versions of those extensions have access to the tsserver
